src/test/java/com/fasterxml/jackson/databind/deser/enums/EnumSameName4302Test.java

```diff
@@ -78,5 +93,22 @@ void testShouldWork() throws Exception
         assertEquals(q("CAT"),
             MAPPER.writeValueAsString(Setter4302Enum.CAT));
     }
+
+    @Test
+    void testDeserializeSuccessfulUnaffectedField() throws Exception
+    {
+        AsField4302Bean bean = new AsField4302Bean();
+
+        // test serialization
+        assertEquals(
+            a2q("{'some_enum':'APPLE_SUGAR','other_prop':'someOtherField'}"),
+            SNAKE_MAPPER.writeValueAsString(bean));
+
+        // test deserialization
+        AsField4302Bean result = SNAKE_MAPPER.readValue(
+            a2q("{'some_enum':'SOME_PERSON', 'other_prop':'thisField'}"), AsField4302Bean.class);
+        assertEquals(AsField4302Enum.SOME_PERSON, result.someEnum);
+        assertEquals("thisField", result.otherProp);
```

Comment:
This test corresponds to the failing test in ion module --when there is Enum field, we should keep renaming it, same as before

Reply:
Right.